TestRunner.psm1

```diff
@@ -81,84 +81,218 @@ function Invoke-RemoteCommand($server, [PSCredential] $credential, $argumentList
     }
 }
 
-function Log($message) {
+function Write-Log {
+    param(
+        [Parameter(Mandatory=$True)]
+        [string]
+        $Message,
+
+        [Parameter(Mandatory=$True)]
+        [ValidateSet("ERROR", "INFO", "DEBUG")]
+        [string]
+        $Level
+    )
+
     $timestamp = Get-Date -Format "yyyy-MM-dd HH:mm:ss,fff"
-    $output = "[$timestamp] $message"
-    Write-Host $output
+    $output = "[$timestamp] $Level $Message"
+
+    if ($Level -in ("ERROR", "WARNING")) {
+        Write-Error $output
+    } else {
+        Write-Host $output
+    }
+
     $output | Out-File -Encoding UTF8 $testResultsPath\PerformanceTesterLog.txt -Append
 }
 
-function Reset-OdsDatabase($backupFilename) {
```

Comment:
This big chunk is significantly rewritten and appears farther below. I found that it was much too rigid in its assumptions (for example, assuming there is _one and only one_ each of data and log files), and it suffers from potential SQL injection attacks.